relion/wizards.py

```diff
@@ -301,3 +309,82 @@ def show(self, form):
+
+
+class RelionWizLogPickParams(EmWizard):
+    _targets = [(ProtRelionAutopickLoG, ['minDiameter',
+                                         'maxDiameter',
+                                         'threshold'])]
+
+    def show(self, form):
+        autopickProt = form.protocol
+        project = autopickProt.getProject()
+        micSet = autopickProt.getInputMicrographs()
+        micfn = micSet.getFileName()
+        coordsDir = project.getTmpPath(micSet.getName())
+        print("coordsDir: ", coordsDir)
+        params, minDiameter, maxDiameter, threshold = autopickProt._getPickArgs()
+
+        pw.utils.cleanPath(coordsDir)
+        pw.utils.makePath(coordsDir, 'extra')
+        pickerProps = os.path.join(coordsDir, 'picker.conf')
+        from pyworkflow.em.packages.relion.convert import writeSetOfMicrographs
+        micStarFn = os.path.join(coordsDir, 'input_micrographs.star')
+
+        def _postprocessMic(mic, micRow):
+            micFn = mic.getFileName()
+            micBase = os.path.basename(micFn)
+            pw.utils.createLink(micFn, os.path.join(coordsDir, micBase))
+            micRow.setValue(md.RLN_MICROGRAPH_NAME, micBase)
+
+        writeSetOfMicrographs(micSet, micStarFn, postprocessImageRow=_postprocessMic)
+
+        f = open(pickerProps, "w")
+
+        #params = params.replace('--odir ""', '--odir extra')
+        autopickCmd = "%s relion_autopick " % pw.getScipionScript()
+        autopickCmd += ' --i input_micrographs.star '
+        autopickCmd += params
+        autopickCmd += ' --LoG_diam_min %(mind) '
+        autopickCmd += ' --LoG_diam_max %(maxd) '
+        autopickCmd += ' --LoG_adjust_threshold %(threshold) '
+
+        args = {
+            "convert": pw.join('apps', 'pw_convert.py'),
+            'coordsDir': coordsDir,
+            'micsSqlite': micSet.getFileName(),
+            "minDiameter": minDiameter,
+            "maxDiameter": maxDiameter,
+            "threshold": threshold,
+            'projDir': project.getPath(), #autopickProt.getWorkingDir(),
+            "autopickCmd": autopickCmd
+        }
+
+        f.write("""
+        parameters = mind,maxd,threshold
+        mind.value = %(minDiameter)s
+        mind.label = Min. Diameter for LoG filter
+        mind.help = some help
+        maxd.value = %(maxDiameter)s
+        maxd.label = Max. Diameter for LoG filter
+        maxd.help = some help
+        threshold.value =  %(threshold)s
+        threshold.label = Adjust default threshold
+        threshold.help = Lower threshold -> more particles
+        runDir = %(coordsDir)s
+        autopickCommand = %(autopickCmd)s
+        convertCommand = %(convert)s --coordinates --from relion --to xmipp --input  %(micsSqlite)s --output %(coordsDir)s --extra %(coordsDir)s/
+        hasInitialCoordinates = false
+        doPickAll = true
+        """ % args)
+        f.close()
+        process = CoordinatesObjectView(autopickProt.getProject(), micfn, coordsDir, autopickProt,
```

Comment:
File "/lmb/home/gsharov/soft/scipion-em-plugins/scipion-em-relion/relion/wizards.py", line 380, in show
    process = CoordinatesObjectView(autopickProt.getProject(), micfn, coordsDir, autopickProt,
NameError: global name 'CoordinatesObjectView' is not defined

For some reason it doesnt work...

Reply:
Yep, there were some wrong imports...I fixed them now...could you try again? Still LoG picker wizard was not working very well for me...but I can't test further right now.